specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2023-12-10-preview/cloudhsm.json

```diff
@@ -0,0 +1,967 @@
+      "type": "object"
+    },
+    "CloudHsmClusterResource": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/CloudHsmClusterSku",
+          "description": "SKU details"
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Cloud HSM Cluster Resource",
+      "type": "object",
+      "x-ms-azure-resource": true
+    },
+    "CloudHsmClusterProperties": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster's provisioningState",
+          "enum": [
+            "Provisioning",
+            "Succeeded",
+            "Failed",
+            "Deleting",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "autoGeneratedDomainNameLabelScope": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster's auto-generated Domain Name Label Scope"
+        },
+        "securityDomain": {
+          "$ref": "#/definitions/CloudHsmClusterSecurityDomainProperties"
+        },
+        "hsms": {
+          "type": "array",
+          "description": "An array of Cloud HSM Cluster's HSMs",
+          "x-ms-identifiers": [
+            "fqdn"
+          ],
+          "items": {
+            "$ref": "#/definitions/CloudHsmProperties"
+          }
+        },
+        "publicNetworkAccess": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster public network access"
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "description": "List of private endpoint connection resources",
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "statusMessage": {
+          "type": "string",
+          "description": "Cloud HSM Cluster status message",
+          "readOnly": true
+        },
+        "cloudHsmClusterRestoreProperties": {
+          "$ref": "#/definitions/CloudHsmClusterRestoreProperties"
+        }
+      },
+      "description": "Properties of a Cloud HSM Cluster.",
+      "type": "object"
+    },
+    "CloudHsmClusterPatchParameters": {
+      "properties": {
+        "tags": {
+          "type": "object",
+          "description": "The Cloud HSM Cluster's tags",
+          "additionalProperties": {
+            "type": "string",
+            "description": "The Cloud HSM Cluster's tag's additional properties"
+          }
+        },
+        "sku": {
```

Comment:
I dont think we can change SKU? Is that supported ?

Reply:
@sumantshiv, this is not supported today, but it is being like that since the first API version. If we are certain we will never allow the user to update the SKU with a PATCH Operation we can remove it. 
[2022-08-31-preview swagger](https://github.com/Azure/azure-rest-api-specs/blob/4816586e353f83f79e56789b171eb888054824f1/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2022-08-31-preview/cloudhsm.json#L694C1-L711C7)